src/nlpviewer/components/TextViewer.tsx

```diff
@@ -80,139 +84,260 @@ function TextViewer({ plugins, onEvent }: TextViewerProp) {
   const selectedLink = links.find(link => link.id === selectedLinkId) || null;
   const enabledPlugins = plugins.filter(p => p.enabled(appState));
 
-  return (
-    <div className={style.text_viewer}>
-      <main className={style.layout_container}>
-        <div className={style.metadata_side_container}>
-          <TextDetail
-            annotationLegends={annotationLegendsWithColor}
-            linkLegends={linksLegendsWithColor}
-            attributes={attributes}
-            ontology={ontology}
-          />
+  const pluginsByName = new Map<string, IPlugin>(
+    enabledPlugins.map(
+      p => [p.name, p]
+    )
+  );
+
+  var pluginKey = 0
+
+  function renderPlugin(p:IPlugin){
+    const Comp = p.component;
+    pluginKey = pluginKey + 1;
+    return <Comp key={pluginKey} dispatch={dispatch} appState={appState}/>;
+  }
+
+  function renderPluginByName(name: string){
+    if (pluginsByName.has(name)){
+      const p = pluginsByName.get(name);
+      if (typeof p !== 'undefined'){
+        return renderPlugin(p);
+      }
+    } 
+    return null;
+  }
+
+  function renderAllPlugin(){
+    console.log("Rendering all plugins")
+    if (enabledPlugins.length === 0){
+      return (
+        <div className={style.plugins_container}>
+          {enabledPlugins.map((p, i) => {
```

Comment:
when `enabledPlugins.length ===  0` this doesn't seem to be doing anything

Reply:
Now rendering "No plugins".